src/store/eventing/mode.ts

```diff
@@ -0,0 +1,8 @@
+import { Action } from '@reduxjs/toolkit';
+import eventManager from '../../common/EventManager';
+import { AppState } from '../types';
+import { Event } from '../../@types';
+
+export const handleToggleAnnotationModeAction = (prevState: AppState, nextState: AppState, action: Action): void => {
+    eventManager.emit(Event.ANNOTATIONS_MODE_CHANGE, action);
```

Comment:
Should we pull out only the information we want to send? Otherwise we're exposing internal action properties like `type`.

Reply:
Should we be sending the new state mode instead? i.e. the result of the state change?